### ISSUE TITLE
🚧 Dev ➾ Add Support to `taq scaffold` for Checking-Out a Specific Branch of a Scaffold Repo

### ISSUE THREAD

**Issue by hu3man** [ecadlabs/taqueria#1353]
### 🚥 Definition of Done

- [X] 🎯 Planning and Design
- [ ] 🏎️ Dev Implementation (PR merged)
- [ ] ✍️ Release Notes Added
- [ ] 🏆 Product Acceptance

### ⚖️ Acceptance Criteria

✨ Functional: 
- [ ] `taq scaffold` has the means to provide a branch name which will check-out the associated branch on the scaffolded project
- [ ] Providing an incorrect branch name fails gracefully with a helpful error message
 
 📐 Non-Functional: 
- [ ] Additional steps are minimized for the user
- [ ] Existing functionality of `taq scaffold` is unchanged
 

### 🔭 Scope

🙅‍♂️ Out of Scope: 
- Direct work on scaffold projects 
 

### 🧰 Work Description

Currently, there is no way to scaffold a project from a branch other than main. While this is effective for end users, it makes testing the effects of (pre)releases challenging

As part of the release process, required changes to scaffolds are staged in a branch on the scaffold repo's. The intention is that as soon as a release goes out, the branch can be merged and Taqueria will have a consistent version and behaviour throughout the product

In order to support this flow, we need an easy way to do manual testing on un-released changes to scaffolds. To support this, we need to implement the ability to check out a particular branch of a scaffold project. This will greatly simplify the testing/validation process and increase the overall quality of Taqueria as a product

### 🦄 Sizing

1

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hu3man** [comment]
What would be really cool is if this feature automatically updated all installed plugins to the branch-specific versions implicitly

Thought @jevonearth or @mweichert ?

**Comment by hu3man** [comment]
BLOCKED BY: #1342
